src/rust/multi.ninemanga/src/parser.rs

```diff
@@ -0,0 +1,300 @@
+	}
+
+	// parse the chapters list present on manga details page
+	pub fn parse_chapter_list(&self, id: String) -> Result<Vec<Chapter>> {
+		let mut chapters: Vec<Chapter> = Vec::new();
+
+		let url = format!("{}/manga/{}?waring=1", self.base_url, id);
+		let html = Request::new(url.as_str(), HttpMethod::Get)
+			.header("Accept-Language", "es-ES,es;q=0.9,en;q=0.8,gl;q=0.7")
+			.header(
+				"User-Agent",
+				"Mozilla/5.0 (Windows NT 10.0; WOW64) Gecko/20100101 Firefox/75",
+			)
+			.html()
+			.expect("Failed to initialize the request");
+
+		let name = String::from(
+			html.select(".bookface img")
+				.attr("alt")
+				.read()
+				.replace("Manga", "")
+				.replace("Манга", "")
+				.trim(),
+		);
+
+		for chapter in html.select("ul.sub_vol_ul > li").array() {
+			let chapter_node = chapter.as_node().expect("Failed to get the node");
+
+			let raw_title = String::from(
+				chapter_node
+					.select("a")
+					.text()
+					.read()
+					.replace("13610", "")
+					.trim(),
+			);
+
+			let chapter_id = chapter_node
+				.select("a.chapter_list_a")
+				.attr("href")
+				.read()
+				.replace(".html", "-10-1.html");
+
+			let url = chapter_id.clone();
+			let chapter_number = extract_f32_from_string(raw_title.clone(), name.clone());
+			let date_updated = get_date(chapter_node, self.date_format, self.date_locale);
+
+			chapters.push(Chapter {
+				id: chapter_id,
+				volume: -1.0,
+				chapter: chapter_number,
+				date_updated,
+				url,
+				lang: self.language.clone(),
+				..Default::default()
+			});
+		}
+
+		Ok(chapters)
+	}
+
+	//parse the maga chapter images list
+	pub fn parse_page_list(&self, _id: String, chapter_id: String) -> Result<Vec<Page>> {
+		let mut pages: Vec<Page> = Vec::new();
+		let mut at = 0;
+
+		let pages_arr = get_chapter_pages(self.base_url.clone(), chapter_id);
+		for url in pages_arr {
+			let html = Request::new(&url, HttpMethod::Get)
+				.header("Accept-Language", "es-ES,es;q=0.9,en;q=0.8,gl;q=0.7")
+				.header(
+					"User-Agent",
+					"Mozilla/5.0 (Windows NT 10.0; WOW64) Gecko/20100101 Firefox/75",
+				)
+				.header("Cookie", "ninemanga_webp_valid=true")
+				.html()
+				.expect("Failed to initialize the request");
+
+			for page in html.select(".pic_box img.manga_pic").array() {
+				let page_node = page.as_node().expect("Failed to get the node");
+				let page_url = page_node.attr("src").read();
+				pages.push(Page {
+					index: at,
+					url: page_url,
+					base64: String::new(),
+					text: String::new(),
+				});
+				at += 1;
+			}
+		}
+
+		Ok(pages)
+	}
+
+	pub fn modify_image_request(&self, request: Request) {
+		request
+			.header("Referer", self.base_url.as_str())
+			.header("Accept-Language", "es-ES,es;q=0.9,en;q=0.8,gl;q=0.7")
+			.header(
+				"User-Agent",
+				"Mozilla/5.0 (Windows NT 10.0; WOW64) Gecko/20100101 Firefox/75",
+			);
+	}
+
+	pub fn handle_url(&self, url: String) -> Result<DeepLink> {
+		Ok(DeepLink {
+			manga: Some(Self::parse_manga_details(self, url)?),
```

Comment:
you changed id so it shouldn't be a url any longer, you need to parse the id out of it

Reply:
still not addressed